scipy/io/arff/tests/test_arffread.py

```diff
@@ -104,10 +104,14 @@ def test_nodata(self):
         # Reading it should result in an array with length 0.
         nodata_filename = os.path.join(data_path, 'nodata.arff')
         data, meta = loadarff(nodata_filename)
-        expected_dtype = np.dtype([('sepallength', '<f8'),
-                                   ('sepalwidth', '<f8'),
-                                   ('petallength', '<f8'),
-                                   ('petalwidth', '<f8'),
+        if sys.byteorder == 'big':
+            end = '>'
+        elif sys.byteorder == 'little':
+            end = '<'
```

Comment:
Just in case, can you add `else: # raise a runtimeerror`? Or just change the `elif ...` to `else:`

Reply:
I think I've done what you asked now. I rebased as well just in case.